### ISSUE TITLE
goTryRaw types are not developer friendly

### ISSUE THREAD

**Issue by IlyaSemenov** [thelinuxlich/go-go-try#3]
Consider this:

```ts
async function fn(a: number) {
  if (a < 0) {
    throw new MyError()
  }
  return otherFn(a) // known to return Promise<ComplexType>
}

const [err, data] = await goTryRaw(fn(123))
```

- `data` is typed as `ComplexType`, which is good.
- `err` is typed as `unknown`, which is inconvenient.

Now as I know that `fn` throws `MyError`, I want to type the return value:

```ts
const [err, data] = await goTryRaw<MyError>(fn(123))
```

- `err` is typed as `MyError`
- `data` is typed as `unknown`, which is inconvenient.

Sometypes `ComplexType` is not even exported as it may be a complex generic created inside a third party library, so it couldn't even be retrieved with `ReturnType<...>` escape hatch.

So basically developer is supposed to use `goTryRaw` without arguments and then cast error to proper type.

Ideally, there should be a way to provide error type but automatically infer return value type. I am not sure that elegant solution to this even exists, but wanted to report this concern nevertheless.

**Comment by thelinuxlich** [comment]
Hey @IlyaSemenov I agree with you! But it's not possible until TS adds this feature: https://github.com/microsoft/TypeScript/issues/26242

**Comment by thelinuxlich** [comment]
What you can do is pass a second generic parameter with `ComplexType`:

```ts
const [err, data] = await goTryRaw<MyError, ComplexType>(fn(123))
```
